os/include/tinyara/pm/pm.h

```diff
@@ -292,6 +292,25 @@ extern "C" {
 void pm_driver_register(void);
 
 #ifdef CONFIG_PM
+
+/****************************************************************************
+ * Name: pm_start
+ *
+ * Description:
+ *   This function is called by the application thread to start the Power
+ *   Management system. This fucntion sets the ready_to_state_change flag which
+ *   enables pm to transition between low and high power states.
+ *
+ * Input parameters:
```

Comment:
Why this interface is required? Can you specify the needs of it?

Reply:
Previously we were using `PM_IDLE_DOMAIN` to suspend PM state transition in `os_start`. Since we are suspending `PM_IDLE_DOMAIN`, so to enable pm state transition, we were using `system_pm_start` which resume the `PM_IDLE_DOMAIN`. This was the temporary fix.

Now with this interface we can avoid the sharing `PM_IDLE_DOMAIN` to `os_start` and `system_pm_start`. It gives better encapsulation to PM Module.